public/app/plugins/datasource/loki/streaming.test.ts

```diff
@@ -0,0 +1,10 @@
+import { getLiveStreamKey } from './streaming';
+
+describe('loki streaming', () => {
+  // jsdom does not have TextEncoder :_(
+  it.skip('key only depends on expression', async () => {
```

Comment:
TextEncoder does not exist in jsdom --- many pretty hacky solutions, none look great.  any ideas?

Reply:
@ryantxu ... no good ideas :) 

i tried mocking `TextEncoder` in the test, which worked, but then i found out that `crypto` is also not support in `jsdom`.. 

(honestly, is "jsdom" abandoned? there's a feature-request issue for `crypto` from 2016)

what we could do, is to extract the hashing part into a separate function, like:

```typescript
export async function hashDigest(str: string): Promise<string> {
  const msgUint8 = new TextEncoder().encode(str); // encode as (utf-8) Uint8Array
  const hashBuffer = await crypto.subtle.digest('SHA-1', msgUint8); // hash the message
  const hashArray = Array.from(new Uint8Array(hashBuffer.slice(0, 8))); // first 8 bytes
  return hashArray.map((b) => b.toString(16).padStart(2, '0')).join('');
}
```

move it to a separate file, and then we could mock it in `streaming.test.ts`

.. or we just drop `streaming.test.ts` for now